swan-lake/development-tutorials/source-code-dependencies/customizing-formatting.md

```diff
@@ -0,0 +1,388 @@
+active: customizing-formatting
+intro: This feature allows developers to customize the formatting of Ballerina projects. This is done with the use of a configuration file. This configuration file can be a local or a remote file, which allows for consistency in code style across projects and simplifies the process of enforcing formatting standards. This is introduced as an experimental feature in Ballerina 2201.9.0.
+---
+
+## Configuring
+
+Users can provide custom formatting configurations with the use of a separate configuration file of the `toml` format. Users can provide an arbitrary name for the file.
+
+The formatting configuration file can be specified in two ways:
+
+* `Ballerina.toml`: Define the path to a configuration file within the Ballerina.toml itself. The path can be absolute, relative to the package root, or a URL (HTTP/HTTPS). This path is specified under the `format` section in `Ballerina.toml` with the use of the `configPath` attribute. 
+
+    ```
+    [format]
+    configPath = "Format.toml"
+    ```
+
+* Format.bal: If no configuration path is specified in Ballerina.toml, the formatter will look for a file named `Format.toml` in the current package root directory.
+* If the `configPath` argument is not provided in the `Ballerina.toml` or the `Format.toml` file does not exist in the project root, default formatting options will be applied.
+* If the format config path is an HTTP/HTTPS URL, a cache of the configuration file will be maintained inside the target directory. If the remote configuration file is modified, then make sure to delete the cached `Format.toml` file inside the `target/format/` directory.
+
+
+## Formatting Options
+
+### Indent
+
+#### indentSize
+
+* Specifies the number of spaces for each level of indentation.
+* Value: Integer (e.g., 2, 4)
+* Default: 4
+
+    ```ballerina
+    function main() {
+        if (condition) {
+            io:println("Indented code");
+        }
+    }
+    ```
+
+#### continuationIndentSize
+
+* Determines the indentation size for continuation lines.
+* Value: Integer (e.g., 4, 8)
+* Default: 8
+
+    ```ballerina
+    function longFunctionName(int param1, int param2, int param3,
+            int param4, int param5) {
+    }
+    ```
+
+### Wrapping
+
+#### maxLineLength
+
+* Maximum line length before code is wrapped.
+* Value: Integer (e.g., 80, 120)
+* Default: No line wrapping if maxLineLength not specified.
+
+#### simpleBlocksInOneLine 
+
+* Preserve single-line blocks in user code.
+* Value: Boolean
+* Default: false
+
+    ```ballerina
+    if x == 1 { return true; }
+    while i < 5 { i += 1; }
+    ```
+
+#### simpleFunctionsInOneLine
+
+* Preserve single-line methods on a single line.
+* Value: Boolean
+* Default: false
+
+    ```ballerina
+    function getName() returns string { return self.name; }
+    ```
+
+### Braces
+
+#### classBraceStyle: 
+
+* Placement of the opening brace of the class definition.
+* Values: NewLine, EndOfLine
```

Comment:
Shouldn't we use backticks for NewLine, EndOfLine etc? @MaryamZi

Reply:
Yeah, since these are fixed values, we should.